src/sources/opentelemetry/mod.rs

```diff
@@ -67,12 +68,12 @@ pub struct OpentelemetryConfig {
 /// Configuration for the `opentelemetry` gRPC server.
 #[configurable_component]
 #[derive(Clone, Debug)]
+#[serde(default = "default_grpc_config")]
```

Comment:
FWIW, I'd drop this (and the one on `HttpConfig`) and just do a manual impl of `Default` for `GrpcConfig`/`HttpConfig` -- using the contents of `default_grpc_config` and `default_http_config`, respectively -- and then use `#[serde(default)]` on the `grpc` and `http` fields.

This is a little more concise and shows that the default for those fields in `OpentelemetryConfig` should derive from the default value of `GrpcConfig` and `HttpConfig` themselves, and then the bit about implementing `Default` is simply, again, a little more concise and common/intuitive.